client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -341,9 +343,14 @@ private void throwIfClosed() throws TxnFailedException {
                 RuntimeException::new);
         UUID txnId = txnSegments.getTxnId();
         Map<Segment, SegmentTransaction<Type>> transactions = new HashMap<>();
+        DelegationTokenProvider tokenProvider = null;
         for (Segment s : txnSegments.getSteamSegments().getSegments()) {
-            SegmentOutputStream out = outputStreamFactory.createOutputStreamForTransaction(s, txnId,
-                    config, txnSegments.getSteamSegments().getDelegationToken());
+            if (tokenProvider == null) {
+                // We can share token providers across segments as delegation tokens are scope + stream specific.
```

Comment:
The eventStreamWriter is specific to a given scope and stream. Can we create one token provider per EventStreamWriter ?

Reply:
Sure, I've made the change as suggested in the latest commit I've pushed.